src/plugin.js

```diff
@@ -0,0 +1,241 @@
+      player.error(null);
+    }
+  };
+
+  let listeners = [];
+
+  // clear any previously registered listeners
+  const cleanup = function() {
+    let listener;
+
+    while (listeners.length) {
+      listener = listeners.shift();
+      player.off(listener[0], listener[1]);
+    }
+    window.clearTimeout(monitor);
+  };
+
+  // creates and tracks a player listener if the player looks alive
+  const healthcheck = function(type, fn) {
+    let check = function() {
+      // playback isn't expected if the player is paused, shut
+      // down monitoring
+      if (player.paused()) {
+        return cleanup();
+      }
+      // playback isn't expected once the video has ended
+      if (player.ended()) {
+        return cleanup();
+      }
+      fn.call(this);
+    };
+
+    player.on(type, check);
+    listeners.push([type, check]);
+  };
+
+  player.on('play', function() {
+    let lastTime = 0;
+
+    cleanup();
+
+    // if no playback is detected for long enough, trigger a timeout error
+    resetMonitor();
+    healthcheck('timeupdate', function() {
+      let currentTime = player.currentTime();
+
+      if (currentTime !== lastTime) {
+        lastTime = currentTime;
+        resetMonitor();
+      }
+    });
+    healthcheck('progress', resetMonitor);
+  });
+
+  player.on('dispose', function() {
+    cleanup();
+  });
+};
+
+// Setup Custom Error Conditions
+const initCustomErrorConditions = function(player, options) {
+
+  // PLAYER_ERR_TIMEOUT
+  monitorPlayback(player, options);
+
+  // PLAYER_ERR_NO_SRC
+  player.on('play', function() {
+    if (!player.currentSrc()) {
+      player.error({
+        code: -1,
+        type: 'PLAYER_ERR_NO_SRC'
+      });
+    }
+  });
+};
+
+/**
+ * Function to invoke when the player is ready.
+ *
+ * This is a great place for your plugin to initialize itself. When this
+ * function is called, the player will have its DOM and child components
+ * in place.
+ *
+ * @function onPlayerReady
+ * @param    {Player} player
+ * @param    {Object} [options={}]
+ */
+const onPlayerReady = (player, options) => {
+
+  player.addClass('vjs-errors');
+
+  // Add to the error dialog when an error occurs
+  player.on('error', function() {
+    let display;
+    let details = '';
+    let error = player.error();
+    let content = document.createElement('div');
+
+    // In the rare case when `error()` does not return an error object,
+    // defensively escape the handler function.
+    if (!error) {
+      return;
+    }
+
+    error = videojs.mergeOptions(error, options.errors[error.code || 0]);
+
+    if (error.message) {
+      details = `<div class="vjs-errors-details">${player.localize('Technical details')}
+        : <div class="vjs-errors-message">${player.localize(error.message)}</div>
+        </div>`;
+    }
+
+    display = player.errorDisplay;
+
+    content.className = 'vjs-errors-dialog';
+    content.innerHTML =
+      `<button class="vjs-errors-close-button"></button>
```

Comment:
This can probably be removed - as well as anything related to `closeButton`. The `ModalDialog` component has a built-in close button. It's not active on the `errorDisplay` by default (by virtue of the `uncloseable: true` option), but if you call `errorDisplay.closeable(true)` it will become closeable.